pkg/api/v1/types.go

```diff
@@ -2110,6 +2110,8 @@ type PodSpec struct {
 	// If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	// SchedulingMismatchedPredicateResults Summary of false podFitsOnNode predicates for investigating unscheduled pods.
```

Comment:
- is the API conventions statement debatable? After all `map[string]int` seems to match the convention for `NodeSelector` .  If its a hard requirement then, ok, I can create a struct to wrap the data, but its largely unnecessary for this use case.
- Related to the last question --- what is `[]PredicateResult` ? I guess maybe you're referring to a new struct that I could create?